PolyEngine/API/Src/pe/api/deps/DependencyManager.cpp

```diff
@@ -0,0 +1,124 @@
+#include <pe/api/IEngine.hpp>
+
+#include <pe/api/deps/DependencyManager.hpp>
+#include <pe/core/utils/Logger.hpp>
+
+using ::pe::core::utils::gConsole;
+
+namespace pe::api::deps {
+
+DependencyManager& DependencyManager::get()
+{
+	static DependencyManager instance;
```

Comment:
what's the advantage of static instance compared to static private field in class?

Reply:
This is a Meyers-singleton pattern. The main advantage is that the DependencyManager will be lazily constructed when the `get()` is called instead of constructed during static initialization. It may not even be constructed if there is no need for it.